WcaOnRails/spec/models/competition_spec.rb

```diff
@@ -269,6 +269,22 @@
       expect(competition).to be_valid
       expect(competition.warnings_for(nil)[:events]).to eq "Please add at least one event before confirming the competition."
     end
+
+    it "does not warn about other different championships" do
+      different_championship_type = FactoryBot.create :competition, :confirmed, :visible, starts: Date.new(2019, 05, 06), championship_types: ["_North America"]
+      different_year = FactoryBot.create :competition, :confirmed, :visible, starts: Date.new(2018, 02, 03), championship_types: ["world"]
+      competition = FactoryBot.create :competition, starts: Date.new(2019, 10, 01), championship_types: ["world"]
+      expect(competition).to be_valid
+      expect(competition.warnings_for(nil)["world"]).to eq nil
+    end
+
+    it "warns if championship already exists" do
+      existing_competition = FactoryBot.create :competition, :confirmed, :visible, showAtAll: true, starts: Date.new(2019, 05, 06), championship_types: ["world", "_Oceania"]
```

Comment:
Nit: `:visible` sets `showAtAll: true` for you:
https://github.com/thewca/worldcubeassociation.org/blob/e909979644941d6bb1dbdef02769b7a76e074b81/WcaOnRails/spec/factories/competitions.rb#L108-L111